src/arrayMethodJoin.js

```diff
@@ -5,8 +5,26 @@
  */
 function applyCustomJoin() {
   [].__proto__.join2 = function(separator) {
-    // write code here
+    let result = '';
+
+    let sep = separator;
+
+    if (sep === undefined) {
+      sep = ',';
+    }
+
+    for (let i = 0; i < this.length; i++) {
+      if (this[i] !== null && this[i] !== undefined) {
```

Comment:
The original join function also excludes empty elements (undefined, null, and empty strings). You need to add a check for empty strings too.